web/src/components/organisms/Settings/Integration/hooks.ts

```diff
@@ -0,0 +1,152 @@
+      .map<IntegrationMember | undefined>(member =>
+        member && member.__typename === "WorkspaceIntegrationMember" && member.integration
+          ? {
+              active: member.active,
+              integration: fromIntegration(member.integration),
+              integrationRole: member.integrationRole,
+              invitedById: member.invitedById,
+            }
+          : undefined,
+      )
+      .filter(
+        (integrationMember): integrationMember is IntegrationMember =>
+          !!integrationMember &&
+          integrationMember.integration.name.toLowerCase().includes(searchTerm ?? ""),
+      );
+  }, [workspace, searchTerm]);
+
+  const handleIntegrationConnectModalClose = useCallback(() => {
+    SetSelectedConnectionModalIntegration(undefined);
+    setIntegrationConnectModalShown(false);
+  }, []);
+
+  const handleIntegrationConnectModalOpen = useCallback(() => {
+    SetSelectedConnectionModalIntegration(undefined);
+    setIntegrationConnectModalShown(true);
+  }, []);
+
+  const handleIntegrationSettingsModalClose = useCallback(() => {
+    setIntegrationSettingsModalShown(false);
+  }, []);
+
+  const handleIntegrationSettingsModalOpen = useCallback((integrationMember: IntegrationMember) => {
+    console.log(integrationMember);
+
+    SetSelectedIntegrationMember(integrationMember);
+    setIntegrationSettingsModalShown(true);
+  }, []);
+
+  const handleConnectionModalIntegrationSelect = useCallback((integration: Integration) => {
+    SetSelectedConnectionModalIntegration(integration);
+  }, []);
+
+  const [addIntegrationToWorkspaceMutation] = useAddIntegrationToWorkspaceMutation();
+
+  const handleIntegrationConnect = useCallback(async () => {
+    if (!selectedConnectionModalIntegration || !workspaceId) return;
+    const integration = await addIntegrationToWorkspaceMutation({
+      variables: {
+        integrationId: selectedConnectionModalIntegration.id,
+        workspaceId,
+        role: GQLRole.Reader,
+      },
+    });
+    if (integration.errors || !integration.data?.addIntegrationToWorkspace) {
+      setIntegrationConnectModalShown(false);
+    }
+    setIntegrationConnectModalShown(false);
+    refetch();
+  }, [addIntegrationToWorkspaceMutation, selectedConnectionModalIntegration, workspaceId, refetch]);
+
+  const [updateIntegrationToWorkspaceMutation] = useUpdateIntegrationOfWorkspaceMutation();
+
+  const handleUpdateIntegration = useCallback(
+    async (role: Role) => {
+      if (!workspaceId || !selectedIntegrationMember) return;
+      const integration = await updateIntegrationToWorkspaceMutation({
+        variables: {
+          integrationId: selectedIntegrationMember?.integration.id,
+          workspaceId,
+          role: role as GQLRole,
+        },
+      });
+      if (integration.errors || !integration.data?.updateIntegrationOfWorkspace) {
+        setIntegrationConnectModalShown(false);
+      }
+      setIntegrationConnectModalShown(false);
```

Comment:
Again, redundant. Please be careful when copy & pasting that the original code is good. Please update anywhere else its like this

Reply:
https://github.com/reearth/reearth-cms/blob/58c8be6313ec45a489b63534dab11973f18ce352/web/src/components/organisms/Settings/Integration/hooks.ts#L121-L127